README.md

```diff
@@ -193,54 +214,204 @@ The method `evaluate_quadratic_expression` has the following interface:
+
+### Predefined Unsigned Integers
+
+BigNum supports operations over unsigned integers, with predefined types for 256, 384, 512, 768, 1024, 2048, 4096 and 8192 bit integers. Keep in mind these are not Fields.
+
+All arithmetic operations are supported including integer div and mod functions (make sure to use udiv, umod). Bit shifts and comparison operators are not yet implemented.
+
+```rust
+use dep::bignum::fields::U256::U256Params;
+use dep::bignum::BigNum;
+
+type U256 = BigNum<3, 257, U256Params>;
 
-For other moduli (e.g. those used in RSA verification), both `modulus` and `redc_param` must be computed and formatted according to the following speficiations:
+fn foo(x: U256, y: U256) -> U256 {
+    x.udiv(y)
+}
+```
 
-`modulus` represents the BigNum modulus, encoded as an array of `Field` elements that each encode 120 bits of the modulus. The first array element represents the least significant 120 bits.
+### Predefined Fields
 
-`redc_param` is equal to `(1 << (2 * Params::modulus_bits())) / modulus` . This must be computed outside of the circuit and provided either as a private witness or hardcoded constant. (computing it via an unconstrained function would be very expensive until noir witness computation times improve)
+This library includes the follow field presets:
+
+| Curve       | Base Field Instance                | Scalar Field Instance              |
+|-------------|------------------------------------|------------------------------------|
+| [BLS12-377](https://eprint.iacr.org/2020/351)   | `BLS12_377_Fq_Params`            | `BLS12_377_Fr_Params`            |
+| [BLS12-381](https://electriccoin.co/blog/new-snark-curve/)   | `BLS12_381_Fq_Params`            | `BLS12_381_Fr_Params`            |
+| [BN254](https://zips.z.cash/protocol/protocol.pdf)   | `BN254_Fq_Params`                    | (Scalar field is the native field in Noir) |
+| [Curve25519](https://cr.yp.to/ecdh/curve25519-20060209.pdf)  | `ED25519_Fq_Params`              | `ED25519_Fr_Params`              |
+| [MNT4-753](https://eprint.iacr.org/2014/595)    | `MNT4_753_Fq_Params`             | `MNT4_753_Fr_Params`             |
+| [MNT6-753](https://eprint.iacr.org/2014/595)    | `MNT6_753_Fq_Params`             | `MNT6_753_Fr_Params`             |
+| [Pallas](https://electriccoin.co/blog/the-pasta-curves-for-halo-2-and-beyond/)*      | `Pallas_Fr_Params` | `Pallas_Fq_Params` |
+| [Secp256k1](https://en.bitcoin.it/wiki/Secp256k1) | `Secp256k1_Fq_Params`            | `Secp256k1_Fr_Params`            |
+| [Secp256r1](https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.186-4.pdf) | `Secp256r1_Fq_Params`            | `Secp256r1_Fr_Params`            |
+| [Secp384r1](https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.186-4.pdf) | `Secp384r1_Fq_Params`            | `Secp384r1_Fr_Params`            |
+
+* TODO: this should probably be adjusted, to avoid confusion
```

Comment:
TODO

Reply:
This is a TODO for your team, it has nothing to do with the documentation itself. I can remove it.